livechat/reports/client.py

```diff
@@ -0,0 +1,457 @@
+
+
+class ReportsAPI33(ReportsApiInterface):
+    ''' Configuration API client in version 3.3 class. '''
+
+    # Chats
+    def agents_chatting_duration(self,
+                                 to: str = None,
+                                 from_param: str = None,
+                                 agents: str = None,
+                                 groups: str = None,
+                                 tags: str = None,
+                                 customer_client_ids: str = None,
+                                 distribution: str = None,
+                                 timezone: str = None,
+                                 tagged: bool = None,
+                                 payload: dict = None,
+                                 headers: dict = None) -> httpx.Response:
+        ''' Shows the average chatting duration of agents within a license.
+
+        Args:
+            to (str): Date in the RFC3339 format, which also contains a timezone.
+                      This timezone will be used if no `timezone` is provided.
+            from_param (str): Date in the RFC3339 format, which also contains a timezone.
+                              This timezone will be used if no `timezone` is provided.
+            agents (str): Agent emails separated by a comma.
+                          If not specified, returns the data for all agents within the license.
+            groups (str): Group IDs separated by a comma.
+            tags (str): Names of tags separated by a comma.
+            customer_client_ids (str): Client IDs separated by a comma.
+            distribution (str): Possible values: `hour`, `day-hours`, `day`, `month`; defaults to `day`.
+            timezone (str): Timezone in the TZ format (e.g. America/Phoenix).
+                            Defaults to the requester's timezone.
+                            If not present, `from` is parsed to get the requester's timezone.
+            tagged (bool): Possible values: `true`, `1`, `false`, `0`
+            payload (dict): Custom payload to be used as request's data.
+                            It overrides all other parameters provided for the method.
+            headers (dict): Custom headers to be used with session headers.
+                            They will be merged with session-level values that are set,
+                            however, these method-level parameters will not be persisted across requests.
+
+        Returns:
+            httpx.Response: The Response object from `httpx` library,
+                            which contains a server’s response to an HTTP request.
+        '''
+        if payload is None:
+            payload = prepare_payload(locals())
+            if from_param is not None:
+                del payload['from_param']
+                payload['from'] = from_param
+        else:
+            if payload['from_param']:
+                payload['from'] = payload['from_param']
+                del payload['from_param']
```

Comment:
```suggestion
        if payload is None:
            payload = prepare_payload(locals())
            if date_from is not None:
                payload['from'] = date_from
            if date_to is not None:
                payload['to'] = date_to
```
I suggest to add `date_to` and `date_from` to protected keys in `prepare_payload` so that it's not necessary to remove them manually each time. Also, `else` part should not be necessary at all. If someone provides payload manually, it's possible to provide correct key (`from`). See my comment for the examples.